finagle-http/src/main/scala/com/twitter/finagle/Http.scala

```diff
@@ -107,6 +107,7 @@ object Http extends Client[Request, Response] with HttpRichClient
     val stack: Stack[ServiceFactory[Request, Response]] =
       StackClient.newStack
         .replace(TraceInitializerFilter.role, new HttpClientTraceInitializer[Request, Response])
+        .replace(StackClient.Role.prepFactory, DelayedRelease.module)
```

Comment:
Leaving prepConn out will mean HTTP clients built using Stack and ClientBuilder have be different – see: https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/Codec.scala#L154.

This comment is more for posterity than recommending the work be done in this PR. Eventually, I think we should reconcile the gap.